iceoryx_posh/include/iceoryx_posh/roudi/roudi_config_json_file_provider.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_ROUDI_ROUDI_CONFIG_JSON_FILE_PROVIDER_HPP
+#define IOX_POSH_ROUDI_ROUDI_CONFIG_JSON_FILE_PROVIDER_HPP
+
+#include "iceoryx_posh/roudi/roudi_config_file_provider.hpp"
+
+#include "iceoryx_posh/roudi/cmd_line_args.hpp"
+
+extern "C" {
+#include "tiny-json.h"
+}
+
+namespace iox
+{
+namespace roudi
+{
+static constexpr char defaultConfigJsonPath[] = "/etc/iceoryx/roudi_config.json";
+
+/// @brief Reader for Roudi Configuration using json config file
+class JsonRouDiConfigFileProvider : public iox::roudi::RouDiConfigFileProvider
+{
+  private:
+    constexpr static uint32_t NUMBER_OF_JSON_NODES = 1024;
```

Comment:
This value seems to be somehow arbitrarily chosen. The TOML implementation does not have such a limit. If you really see the need to limit the stack memory usage, then please put this value into `iceoryx_posh_types.hpp ` and document it properly. If not (which is what I assume), just calculate the value using MAX_NUMBER_OF_MEMPOOLS and MAX_SHM_SEGMENTS, to make sure we neither prevent huge config files nor waste stack memory by using too high values.